node/node.go

```diff
@@ -0,0 +1,49 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+type NexNode struct {
```

Comment:
tiny nit... but I'd like to have an interface for the node, and the node itself returns a `*nexNode` (non-exported struct)